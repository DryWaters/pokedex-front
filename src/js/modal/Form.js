import React from 'react';
import PropTypes from 'prop-types';

import { getPokemonSprite } from '../core/PokemonAPI';

import '../../styles/modal/Form.css';

const propTypes = {
  selected: PropTypes.bool.isRequired,
  name: PropTypes.string.isRequired,
  image_path: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

class Form extends React.Component {
  render () {
    return (
      <div>
        <div className={`form ${this.props.selected ? 'selected' : ''}`} onClick={() => { this.props.onClick() }}>
          <img className="sprite" src={getPokemonSprite(this.props.image_path)} alt={`${this.props.name}-alt`}/>
        </div>
      </div>
    );
  }

}

Form.propTypes = propTypes;

export default Form;
