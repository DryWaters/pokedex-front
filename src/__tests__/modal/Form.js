import React from 'react';
import { shallow } from 'enzyme';

import Form from '../../js/modal/Form';

import ExampleJSON from '../../../local_data/example_detail.json';

describe('Form Component', () => {
  let props;
  beforeEach(() => {
    props = {
      selected: false,
      name: ExampleJSON.forms[0].name,
      image_path: ExampleJSON.forms[0].image_path,
      onClick: (() => {}),
    };
  });

  it('should render Form component without crashing', () => {
    shallow(<Form {...props} />);
  });

  it('should match Form component snapshot', () => {
    const wrapper = shallow(<Form {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

});
