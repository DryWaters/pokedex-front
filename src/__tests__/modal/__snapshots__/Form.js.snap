// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Component should match Form component snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form
    image_path="/sprites/pokemon/large/12.png"
    name="Butterfree"
    onClick={[Function]}
    selected={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        alt="Butterfree-alt"
        className="sprite"
        src="https://pokedex-backend-server.herokuapp.com/sprites/pokemon/large/12.png"
/>,
      "className": "form ",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "Butterfree-alt",
        "className": "sprite",
        "src": "https://pokedex-backend-server.herokuapp.com/sprites/pokemon/large/12.png",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt="Butterfree-alt"
          className="sprite"
          src="https://pokedex-backend-server.herokuapp.com/sprites/pokemon/large/12.png"
/>,
        "className": "form ",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Butterfree-alt",
          "className": "sprite",
          "src": "https://pokedex-backend-server.herokuapp.com/sprites/pokemon/large/12.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
